src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -61,7 +62,7 @@ public LanguageServer(SendToClient sendToClient, IPowerFxScopeFactory scopeFacto
             _sendToClient = sendToClient;
             _scopeFactory = scopeFactory;
 
-            _logger = logger ?? ((string s) => { });
+            _logger = logger;
```

Comment:
>_logger = logger; [](http://example.com/codeflow?start=12&length=17)

Key change
This suppresses the need to compute 's' string

Reply:
Not sure if you want me to do something here but yes that is the intention. Even if we default logger to no-op function and it is not provided, we would be creating strings that wouldn't be logged. These strings contain PII and are expensive to create especially with dumping the whole payload and all suggestions in the completion function. If we default to null and not a dummy/no-op function, then strings would not be created and logged. Both solutions have same effect of not logging anything but the latter saves us from introducing performance degradation for all consumers of language server by design/default. If _logger is provided then strings would be created and logged no matter what we default it to. That said, we should take another look at this. Do we want to include PII information in these log strings? Do we need to emit huge strings just for logging purposes?